core/src/test/kotlin/org/evomaster/core/problem/external/ExternalServiceUtilsTest.kt

```diff
@@ -0,0 +1,49 @@
+package org.evomaster.core.problem.external
+
+import org.evomaster.core.problem.external.service.ExternalServiceUtils.isAddressAvailable
+import org.evomaster.core.problem.external.service.ExternalServiceUtils.nextIPAddress
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import java.net.InetSocketAddress
+import java.net.ServerSocket
+
+
+class ExternalServiceUtilsTest {
+
+    /**
+     * Tests the next IP address generator. If IP address is out of range from
+     * loopback range should throw exception as well as if IP address is given
+     * in wrong format.
+     */
+    @Test
+    fun testNextAddress() {
+        assertEquals("127.1.100.2", nextIPAddress("127.1.100.1"))
+        assertNotEquals("127.0.0.1", nextIPAddress("127.0.0.0"))
+
+        assertThrows(Exception::class.java) {
+            nextIPAddress("127.255.255.254")
```

Comment:
is it due to being out of range? since we check if the ip is avaiable, we might allow the loop, eg, from the begining. how do you thint?

Reply:
Loopback address ranges from 127.0.0.0 to 127.255.255.255. Usually highest number of a range considered as broadcast address, to avoid unexpected conflicts this address is intentionally skipped. That's why this check is there.